unified-prototype/unified-plugin/plugin-plugin/src/integTest/groovy/org/gradle/api/experimental/plugin/JavaGradlePluginPluginSpec.groovy

```diff
@@ -0,0 +1,125 @@
+package org.gradle.api.experimental.plugin
+
+import org.gradle.test.fixtures.AbstractSpecification
+
+/**
+ * Integration tests for the Gradle plugin ecosystem's {@code javaGradlePlugin} plugin project type.
+ * plugin project type.
+ */
+class JavaGradlePluginPluginSpec extends AbstractSpecification {
+    def "gradle java plugin project builds"() {
+        given:
+        settingsFile << """
+            plugins {
+                id("org.gradle.experimental.plugin-ecosystem")
+            }
+            
+            dependencyResolutionManagement {
+                repositories {
+                    mavenCentral()
+                }
+            }
+    
+            rootProject.name = "example-plugin"
+        """
+
+        buildFile << """
+            javaGradlePlugin {
+                description = "An example Gradle plugin writen in Java"
+                
+                dependencies {
+                    implementation("com.google.guava:guava:33.4.0-jre")
+                }
+                
+                gradlePlugins {
+                    gradlePlugin("custom-ecosystem") {
+                        id = "org.gradle.example"
+                        implementationClass = "org.gradle.example.ExamplePlugin"
+                    }
+                }
+            }
+        """
+
+        file("src/main/java/org/gradle/example/ExamplePlugin.java") << defineExamplePlugin()
+
+        expect:
+        succeeds(":build")
+    }
+
+    def "gradle java plugin project can be included and applied"() {
+        given:
+        settingsFile << """
+            pluginManagement {
+                includeBuild("plugin")
+            }
+            
+            include("example-project")
+            
+            rootProject.name = "example-plugin-use"
+        """
+
+        and:
+        file("example-project/build.gradle") << """
+            plugins {
+                id("org.gradle.example")
+            }
+        """
+
+        and:
+        file("plugin/build.gradle.dcl") << """
+            javaGradlePlugin {
+                description = "An example Gradle plugin writen in Java"
+                
+                dependencies {
+                    implementation("com.google.guava:guava:33.4.0-jre")
+                }
+                
+                gradlePlugins {
+                    gradlePlugin("custom-ecosystem") {
+                        id = "org.gradle.example"
+                        implementationClass = "org.gradle.example.ExamplePlugin"
+                    }
+                }
+            }
```

Comment:
From a modeling POV, this is kind of unique when compared to other prototype ecosystems. This is definitely the strongest case of many things in one project, except they all inherently represent a single publication. 

I know this is just following the existing API, but I wonder if we were to do this "for real", if we'd use a more different structure.

e.g.,

```
gradlePlugin {
    dependencies {
        ...
    }
    plugins { 
        // this isn't valid today as a name because of the dots
        id("org.gradle.example.my") {
            // We could have a convention of converting id -> class 
            implementationClass = "org.gradle.example.MyPlugin"
        }
    }
}
```

The multiple "gradlePlugin" in this test look odd. I don't think we have to change this.

Reply:
Good point, this is wordy and redundant.  I've udpated it to this:

```
javaGradlePlugin {
    description = "An example Gradle plugin writen in Java"
    
    dependencies {
        implementation("com.google.guava:guava:33.4.0-jre")
    }
    
    registers {
        id("org.gradle.example") {
            implementationClass = "org.gradle.example.ExamplePlugin"
        }
    }
}
```

What do you think?

Reply:
Would `plugins` work instead of `registers`?

Reply:
It would, but I think part of the confusion to mitigate is the repetition of `plugin` in any form.  This _project_ is used to _register_ new plugins.  I'd rather keep that or use another word that isn't "plugin/s".